src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.StateSet.cs

```diff
@@ -39,10 +39,15 @@ public DiagnosticState GetState(StateType stateType)
                 return _state[(int)stateType];
             }
 
-            public void Remove(object documentOrProjectId)
+            public void Remove(object documentOrProjectId, bool includeProjectState = true)
```

Comment:
can we avoid these types of optional args.  I get very worrid about all the places we don't fix up that i don't know about.